### PR TITLE
Local stored response fetching

### PR DESCRIPTION
**Pull request by zhongshixi**
*Problem*
1. make the local stored response fetcher a separate and independent PR according to the feed back from https://github.com/prebid/prebid-server/pull/3326#issuecomment-1841905156
```
We also think fetcher change should be a separate PR with unit tests, not a part of this PR. I also understand stored responses set up will not work without File System fetcher changes. Maybe fetcher PR should go first.
```

this PR is to unblock the hello world example framework i try to build

2. add unit test coverage for the stored response

### CODE REVIEW

**Comment by zhongshixi** [comment]
cc @bretg for visibility 
cc @VeronikaSolovei9 @guscarreon for code review

**Review by VeronikaSolovei9 — APPROVED** [review]
Looks good to me, it works locally, I can execute requests with stored responses successfully. Test looks good too.

File: stored_requests/backends/file_fetcher/fetcher.go [hunk 1]
```diff
@@ -35,8 +35,10 @@ func (fetcher *eagerFetcher) FetchRequests(ctx context.Context, requestIDs []str
 	return storedRequests, storedImpressions, errs
 }
 
+// Fetch Responses - Implements the interface to read the stored response information from the fetcher's FileSystem, the directory name is "stored_responses"
 func (fetcher *eagerFetcher) FetchResponses(ctx context.Context, ids []string) (data map[string]json.RawMessage, errs []error) {
-	return nil, nil
+	storedResponses := fetcher.FileSystem.Directories["stored_responses"].Files
```
Comment (VeronikaSolovei9) [inline stored_requests/backends/file_fetcher/fetcher.go:40]: Pre Gus suggestion, please use "data", not "storedResponses" https://github.com/prebid/prebid-server/pull/3326#discussion_r1544811287

Comment (zhongshixi) [reply stored_requests/backends/file_fetcher/fetcher.go]: @VeronikaSolovei9 @guscarreon 
have updated the code implementation

**Comment by zhongshixi** [comment]
cc @SyntaxNode for visibility & code review ( sorry i forgot to mention you)

File: stored_requests/backends/file_fetcher/fetcher.go [hunk 2]
```diff
@@ -35,8 +36,15 @@ func (fetcher *eagerFetcher) FetchRequests(ctx context.Context, requestIDs []str
 	return storedRequests, storedImpressions, errs
 }
 
+// Fetch Responses - Implements the interface to read the stored response information from the fetcher's FileSystem, the directory name is "stored_responses"
 func (fetcher *eagerFetcher) FetchResponses(ctx context.Context, ids []string) (data map[string]json.RawMessage, errs []error) {
```
Comment (AlexBVolcy) [inline stored_requests/backends/file_fetcher/fetcher.go:40]: Nitpick: what do you think about renaming `fetcher` here to just be `f`

**Review by guscarreon — COMMENTED** [review]
Should `FetchResponses` throw an error when the parameter `ids` is empty? If so, can we add the check and its corresponding test case in `TestStoredResponseFileFetcher`?

File: stored_requests/backends/file_fetcher/fetcher.go (hunk 2 above)
Comment (zhongshixi) [reply stored_requests/backends/file_fetcher/fetcher.go:40]: @AlexBVolcy i think that would be a different discussion, all the other fetchers, e.g `http_fetcher`, `empty_fetcher and  `db_fetcher` are all using the same `fetcher` as the receiver name

**Comment by zhongshixi** [comment]
> FetchResponses

i do not think so, as it changes the behaviour expectation of the interface. I want to make it consistent with what `FetchRequests` behaves - if you pass empty id, you get empty json

File: stored_requests/backends/file_fetcher/fetcher_test.go [hunk 3]
```diff
@@ -25,6 +27,55 @@ func TestFileFetcher(t *testing.T) {
 	validateImp(t, storedImps)
 }
 
+func TestStoredResponseFileFetcher(t *testing.T) {
+
```
[the remark this answers is not in the document]
Comment (zhongshixi) [reply stored_requests/backends/file_fetcher/fetcher_test.go]: deleted

**Review by VeronikaSolovei9 — APPROVED** [review]

**Comment by zhongshixi** [comment]
thanks @VeronikaSolovei9 for approval
need one more approval to get merged :)

**Review by AlexBVolcy — APPROVED** [review]

**Pull request merged**